Source/Assemblies/NuGetFallbackFolderAssemblyResolver.cs

```diff
@@ -14,17 +14,30 @@ namespace Dolittle.Assemblies
     /// 
     /// </summary>
     /// <remarks>
-    /// Linux / macOS : /usr/local/share/dotnet/sdk/NuGetFallbackFolder/{package path}
-    /// Windows       : C:/Program Files/dotnet/sdk/NuGetFallbackFolder/{package path} 
+    /// macOS : /usr/local/share/dotnet/sdk/NuGetFallbackFolder/{package path}
+    /// Linux : /usr/share/dotnet/sdk/NuGetFallbackFolder/{package path}
+    /// Windows : C:/Program Files/dotnet/sdk/NuGetFallbackFolder/{package path} 
     /// </remarks>
     public class NuGetFallbackFolderAssemblyResolver : ICompilationAssemblyResolver
     {
         /// <inheritdoc/>
         public bool TryResolveAssemblyPaths(CompilationLibrary library, List<string> assemblies)
         {
-            var basePath = RuntimeInformation.IsOSPlatform(OSPlatform.Windows)?
-                            @"c:\Program Files\dotnet\sdk\NuGetFallbackFolder":
-                            "/usr/local/share/dotnet/sdk/NuGetFallbackFolder";
+            string basePath;
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                basePath = @"c:\Program Files\dotnet\sdk\NuGetFallbackFolder";
+            }
+            else if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+            {
+                // default location on Ubuntu 19.04
+                basePath = "/usr/share/dotnet/sdk/NuGetFallbackFolder";
+            }
```

Comment:
How will this behave on machines with Linux but not on Ubuntu 19.04 or newer? That's a small subset of Linuxen.

Reply:
It should be the same location for Ubuntu 18.04 at least too and doing a crude search for the location from the official [dotnet-setup repo](https://github.com/dotnet/core-setup/search?p=1&q=%2Fusr%2Fshare&unscoped_q=%2Fusr%2Fshare) at least seemingly shows that they use `/usr/share` for rpm and deb based installations (or that's my review of the search at a glance).

Reply:
In that case - why has this not been failing consistently for all of us when building on linux?

I am not knowledgeable enough about the Linux ecosystem to review or accept this PR. I summon @einari

Reply:
This is used at Runtime - and our build tool uses this and could potentially have the problem. But it has very limited number of references, just the BCL and Dolittle + Autofac for the most part, while where these folders matter is when you have things that are part of what is considered the "Global Assembly Cache" - (yeah you heard it here, I pointed at the elephant😁). For inshance ASP.net stuff is typically found in one of these.